test-framework/junit-internal/src/main/java/io/quarkus/test/QuarkusUnitTest.java

```diff
@@ -906,6 +906,67 @@ public QuarkusUnitTest withConfigurationResource(String resourceName) {
         return this;
     }
 
+    /**
+     * Adds configuration properties using a text block with "key=value" lines.
+     * Blank lines and lines starting with '#' are ignored.
+     *
+     * Example:
+     * .withConfiguration("""
+     * quarkus.datasource.db-kind=postgresql
+     * quarkus.datasource.username=user
+     * quarkus.datasource.password=pass
+     * """)
+     *
+     * @param configBlock text block with configuration lines
+     * @return self
+     */
+    public QuarkusUnitTest withConfiguration(String configBlock) {
+        Objects.requireNonNull(configBlock, "Configuration text block must not be null");
+
+        configBlock.lines()
+                .map(String::trim)
+                .filter(line -> !line.isEmpty() && !line.startsWith("#"))
+                .forEach(line -> {
+                    int idx = line.indexOf('=');
+                    if (idx <= 0) {
+                        throw new IllegalArgumentException(
+                                "Invalid configuration line: '" + line + "'. Expected format 'key=value'");
+                    }
+                    String key = line.substring(0, idx).trim();
+                    String value = line.substring(idx + 1).trim();
```

Comment:
You can extract this to a method and just pass `overrideConfigKey` or `overrideRuntimeConfigKey` as a function to it to avoid duplicating this logic